hydra-node/test/Hydra/Chain/Direct/Contract/Close.hs

```diff
@@ -139,42 +170,51 @@ healthySignature number = aggregate [sign sk snapshot | sk <- healthySigningKeys
  where
   snapshot = healthySnapshot{number}
 
+healthyContestationDeadline :: UTCTime
+healthyContestationDeadline =
+  addUTCTime
+    (fromInteger healthyContestationPeriodSeconds)
+    (snd healthyCloseUpperBoundPointInTime)
+
+healthyClosedUTxOHash :: BuiltinByteString
+healthyClosedUTxOHash =
+  toBuiltin $ hashUTxO @Tx healthyClosedUTxO
+
+healthyClosedUTxO :: UTxO
+healthyClosedUTxO =
+  genOneUTxOFor somePartyCardanoVerificationKey `generateWith` 42
+
 data CloseMutation
   = MutateSignatureButNotSnapshotNumber
-  | MutateSnapshotNumberButNotSignature
-  | MutateSnapshotToIllFormedValue
+  | -- | Change the resulting snapshot number, this should make the signature
+    -- invalid.
+    MutateSnapshotNumberButNotSignature
+  | -- | This test the case when we have a non-initial utxo hash but the snapshot number is 0
+    MutateSnapshotNumberToLessThanZero
   | MutateParties
   | MutateRequiredSigner
   | MutateCloseUTxOHash
-  | MutateValidityInterval
-  | MutateCloseContestationDeadline
-  | MutateCloseContestationDeadlineWithZero
+  | InfiniteLowerBound
+  | InfiniteUpperBound
+  | -- | See spec: 5.5 rule 4 -> contestationDeadline = upperBound + contestationPeriod
+    MutateContestationDeadline
+  | -- See spec: 5.5. rule 5 -> upperBound - lowerBound <= contestationPeriod
```

Comment:
idk if it is the best idea to put spec references here

Reply:
Where would you put it? I thought moving these comments to where the mutation code is because I realized I was often scrolling from the code to this comment to check where in the spec to look at.

Or are you suggesting to just get rid of any reference to the spec? If so, what would be your intention there?